extensions-core/azure-extensions/src/main/java/org/apache/druid/data/input/azure/AzureStorageAccountInputSource.java

```diff
@@ -0,0 +1,253 @@
+
+  @JsonCreator
+  public AzureStorageAccountInputSource(
+      @JacksonInject AzureEntityFactory entityFactory,
+      @JacksonInject AzureCloudBlobIterableFactory azureCloudBlobIterableFactory,
+      @JacksonInject AzureInputDataConfig inputDataConfig,
+      @JacksonInject AzureAccountConfig azureAccountConfig,
+      @JsonProperty("uris") @Nullable List<URI> uris,
+      @JsonProperty("prefixes") @Nullable List<URI> prefixes,
+      @JsonProperty("objects") @Nullable List<CloudObjectLocation> objects,
+      @JsonProperty("objectGlob") @Nullable String objectGlob,
+      @JsonProperty("properties") @Nullable AzureInputSourceConfig azureInputSourceConfig,
+      @JsonProperty(SYSTEM_FIELDS_PROPERTY) @Nullable SystemFields systemFields
+  )
+  {
+    super(SCHEME, uris, prefixes, objects, objectGlob, systemFields);
+    this.entityFactory = Preconditions.checkNotNull(entityFactory, "AzureEntityFactory");
+    this.azureCloudBlobIterableFactory = Preconditions.checkNotNull(
+        azureCloudBlobIterableFactory,
+        "AzureCloudBlobIterableFactory"
+    );
+    this.inputDataConfig = Preconditions.checkNotNull(inputDataConfig, "AzureInputDataConfig");
+    this.azureInputSourceConfig = azureInputSourceConfig;
+    this.azureAccountConfig = azureAccountConfig;
+    this.azureIngestClientFactory = new AzureIngestClientFactory(
+        azureAccountConfig,
+        azureInputSourceConfig
+    );
+  }
+
+  @JsonIgnore
+  @Nonnull
+  @Override
+  public Set<String> getTypes()
+  {
+    return Collections.singleton(SCHEME);
+  }
+
+  @Override
+  public SplittableInputSource<List<CloudObjectLocation>> withSplit(InputSplit<List<CloudObjectLocation>> split)
+  {
+    return new AzureStorageAccountInputSource(
+        entityFactory,
+        azureCloudBlobIterableFactory,
+        inputDataConfig,
+        azureAccountConfig,
+        null,
+        null,
+        split.get(),
+        getObjectGlob(),
+        azureInputSourceConfig,
+        systemFields
+    );
+  }
+
+  @Override
+  public Object getSystemFieldValue(InputEntity entity, SystemField field)
+  {
+    final AzureEntity azureEntity = (AzureEntity) entity;
+
+    switch (field) {
+      case URI:
+        return azureEntity.getUri().toString();
+      case BUCKET:
+        return azureEntity.getLocation().getBucket();
+      case PATH:
+        return azureEntity.getLocation().getPath();
+      default:
+        return null;
+    }
+  }
+
+  @Override
+  protected AzureEntity createEntity(CloudObjectLocation location)
+  {
+    return entityFactory.create(
+        location,
+        new AzureStorage(azureIngestClientFactory, location.getBucket()),
+        SCHEME
+    );
+  }
+
+  @Override
+  protected CloudObjectSplitWidget getSplitWidget()
+  {
+    class SplitWidget implements CloudObjectSplitWidget
+    {
+      @Override
+      public Iterator<LocationWithSize> getDescriptorIteratorForPrefixes(List<URI> prefixes)
```

Comment:
Does this mean that ingestion load will only be properly split amongst multiple tasks if the user uses the `prefix` form of the inputSource? If so, this is worth documenting. What would it take to enable splitting for the other forms?

Reply:
i don't think so? this implementation looks to be the same as S3InputSource. if uris or objects is specified i believe getObjectSize is used instead